hack/build.sh

```diff
@@ -58,29 +60,29 @@ run() {
   fi
 
   # Fast mode: Only compile and maybe run test
-  if $(has_flag --fast -f); then
+  if has_flag --fast -f; then
```

Comment:
This is not as it is supposed to work: `has_flag` _echos_ `true` or `false` and this should be evaluated as a command. So running it in a subshell is the right way todo this construct. The warnings from the linter are bogus. As it is now, it won't work as `if` would evaluate the return code, which is always 0 (so all flags are set). I would recommend to keep as is or change `has_flag` to have a different semantics (but then, why ?).

Reply:
Shellcheck would be happy with `if eval "$(has_flag --fast -f)"`. Ref: https://www.shellcheck.net/wiki/SC2091

It's a bit tricky in this case with `has_flag` design. But I'd opt for adding eval rather than rewriting the function. IMO it's beyond my initial intention to get rid off `shellcheck` warnings.

Indeed the `has_flag` parts are failing current integration tests execution, because of the flags must be evaluated bug @rhuss described above.

Reply:
tbh, I would prefer then to adapt `has_flag` to return 0 or 1 and then keep that `$(has_flag ...)` syntax (instead of adding the eval). The whole purpose of the original form was that is was just shorter. Adding an `eval` would counteract this.

Reply:
@rhuss 
If we adapt has_flag return 0 or 1 like below 

```bash
has_flag() {
    filters="$*"
    for var in "${ARGS[@]}"; do
        for filter in $filters; do
          if [ "$var" = "$filter" ]; then
              return 0
          fi
        done
    done
    return 1
}
```

this will work
```bash
if has_flag --help -h; then
    display_help
    exit 0
fi
```


Do we still need keep that $(has_flag ...) syntax ?

Reply:
nope, it's perfectly fine to change completely to your proposal. Thanks!